root/usr/lib64/ruby/vendor_ruby/discovery/menu.rb

```diff
@@ -136,17 +146,25 @@ def cleanup
 end
 include FastGettext::Translation
 
-def main_loop
+def main_loop(args)
   Signal.trap("TERM") { cleanup }
   Signal.trap("INT") { cleanup }
   Signal.trap("HUP") do
     Newt::Screen.refresh
   end
 
+  pipeline = Pipeline.instance
+  pipeline.data.proxy_url = cmdline('proxy.url')
+  pipeline.data.proxy_type = cmdline('proxy.type')
```

Comment:
This is the pipeline, nothing else but a simple hash. Basically a global variable passed around. That's the design, not bug :-)